OGP1617-Asteroids/src/asteroids/model/Ship.java

```diff
@@ -347,6 +362,74 @@ public static boolean isValidRadius(double radius){
 	private final double radius;
 	
 	/**
+	 * Return the massDensity of this Ship.
+	 */
+	@Basic @Raw
+	public double getMassDensity() {
+		return this.massDensity;
+	}
+	
+	/**
+	 * Return the lowest possible massDensity for this ship.
+	 * 
+	 * @return 
+	 * 			| 1.42 * Math.pow(10, 12)
+	 */
+	@Basic
+	public static double getLowestMassDensity() {
+		return 1.42 * Math.pow(10, 12);
+	}
+	
+	/**
+	 * Check whether the given massDensity is a valid massDensity for
+	 * any Ship.
+	 *  
+	 * @param  massDensity
+	 *         The massDensity to check.
+	 * @return 
+	 *       | result == (massDensity >= 1.42 * Math.pow(10, 12))
+	*/
+	public static boolean isValidMassDensity(double massDensity) {
+		return (massDensity >= getLowestMassDensity());
+	}
+	
+	/**
+	 * Set the massDensity of this Ship to the given massDensity.
+	 * 
+	 * @param  massDensity
+	 *         The new massDensity for this Ship.
+	 * @post   If the given massDensity is a valid massDensity for any Ship,
+	 *         the massDensity of this new Ship is equal to the given
+	 *         massDensity. 
+	 *         Else it will be set to the lowest ThrustForce there is.
+	 *       | if (isValidMassDensity(massDensity))
+	 *       |   then new.getMassDensity() == massDensity
+	 *       | this.massDensity == getLowestMassDensity()
+	 */
+	@Raw
+	public void setMassDensity(double massDensity) {
+		if (isValidMassDensity(massDensity))
+			this.massDensity = massDensity;
+		else{
+		this.massDensity = getLowestMassDensity();
+		}
+	}
+	

```

Comment:
In de specificatie staat er de else clause er niet, ik had ook gehoord dat je zon twee gevallen best in 2 postcondities steekt dus ne post voor isValid... en !isValid...